src/shared/dissect-image.c

```diff
@@ -1409,56 +1414,101 @@ int decrypted_image_relinquish(DecryptedImage *d) {
         return 0;
 }
 
-int root_hash_load(const char *image, void **ret, size_t *ret_size) {
+int verity_metadata_load(const char *image, void **roothash, size_t *roothash_size, char **hashdevice) {
         _cleanup_free_ char *text = NULL;
         _cleanup_free_ void *k = NULL;
         size_t l;
         int r;
 
         assert(image);
-        assert(ret);
-        assert(ret_size);
 
         if (is_device_path(image)) {
                 /* If we are asked to load the root hash for a device node, exit early */
-                *ret = NULL;
-                *ret_size = 0;
+                if (roothash)
+                        *roothash = NULL;
+                if (roothash_size)
+                        *roothash_size = 0;
+                if (hashdevice)
+                        *hashdevice = NULL;
                 return 0;
         }
 
-        r = getxattr_malloc(image, "user.verity.roothash", &text, true);
-        if (r < 0) {
-                char *fn, *e, *n;
+        if (roothash) {
+                assert(roothash_size);
 
-                if (!IN_SET(r, -ENODATA, -EOPNOTSUPP, -ENOENT))
-                        return r;
+                r = getxattr_malloc(image, "user.verity.roothash", &text, true);
+                if (r < 0) {
+                        char *fn, *e, *n;
 
-                fn = newa(char, strlen(image) + STRLEN(".roothash") + 1);
-                n = stpcpy(fn, image);
-                e = endswith(fn, ".raw");
-                if (e)
-                        n = e;
+                        if (!IN_SET(r, -ENODATA, -EOPNOTSUPP, -ENOENT))
+                                return r;
 
-                strcpy(n, ".roothash");
+                        fn = newa(char, strlen(image) + STRLEN(".roothash") + 1);
+                        n = stpcpy(fn, image);
+                        e = endswith(fn, ".raw");
+                        if (e)
+                                n = e;
+                        else {
+                                e = endswith(fn, ".squashfs");
```

Comment:
hmm, this looks awfully specific to the contents. So far we stuck to `.raw` which is a nicely generic suffix.

Reply:
Ok, will remove